sycl/source/kernel_bundle.cpp

```diff
@@ -115,14 +115,12 @@ bool kernel_bundle_plain::is_specialization_constant_set(
 
 const std::vector<device>
 removeDuplicateDevices(const std::vector<device> &Devs) {
-  auto compareDevices = [](device a, device b) {
-    return getSyclObjImpl(a)->getHandleRef() <
-           getSyclObjImpl(b)->getHandleRef();
-  };
+  auto compareDevices = [](device a, device b) { return a != b; };
```

Comment:
Even if this does work, I believe C++ specification requires this function to establish strict weak ordering (compareDevices(a, b) == true => compareDevices(b, a) == false, https://en.cppreference.com/w/cpp/named_req/Compare), which this one doesn't do. I think we should switch to comparing Impl pointers here.

Reply:
thank you, updated in this commit https://github.com/intel/llvm/pull/6880/commits/b662fd563b9cc25ca47122722ab5c335652fd8ea